doc/bfast/fixed_angle_broadband_simulations_in_Meep.md

```diff
@@ -0,0 +1,154 @@
+of the incident wave can be fixed over a broad frequency spectrum. This
+requires the addition of a new field. It is assumed that the reader is
+already familiar with the UPML formulation in MEEP [3], from which
+the equations will be modified.  B. Liang et al applied this method by
+splitting the electric and magnetic fields into multiple terms. This is
+different from the UPML formulation in MEEP, which allows simulation of
+more complex media, and so the method of B. Liang et al has to be adapted.
+
+Boundary conditions
+===================
+
+The fields from section 3 of *Notes on the UPML implementation in MEEP*
+[3] are first redefined as:
+```math
+\text{field}'(x,y,z) = \text{field}(x,y,z)e^{-i(k_{x}x+k_{y}y)}, \tag{1}
+```
+where $k_{x}$ and $k_{y}$ are the wave vector components in the x and y
+directions. This is for a structure which is periodic in these
+directions. Taking the electric field $E$ as an example, the new
+boundary condition can be expressed as
+```math
+E'(x+a,y+b,z) = E(x+a,y+b,z)e^{-i(k_{x}(x+a)+k_{y}(y+b))} \tag{2}
+```
+where a is the length of the unit cell in the x direction and b in the y direction.
+Substituting in the original Bloch periodic boundary conditions gives
+```math
+E'(x+a,y+b,z) = E(x,y,z)e^{i(k_{x}a+k_{y}b)}e^{-i(k_{x}(x+a)+k_{y}(y+b))}. \tag{3}
+```
+Cancelling the $a$ and $b$ terms gives
+```math
+E'(x+a,y+b,z) =E(x,y,z)e^{-i(k_{x}x+k_{y}y)}=E'(x,y,z), \tag{4}
+```
+and so the boundary conditions are now periodic.
+
+Formulation
+===========
+
+Equation (5) from section 3 of *Notes on the UPML implementation in
+MEEP* [3] is
+```math
+\vec{K} = \nabla \times \vec{H}=-i\omega(1+\frac{i\sigma_{D}}{\omega})\vec{C}, \tag{5}
+```
+where $\vec{H}$ is the magnetic field, $\sigma_{D}$ the conductivity and
+$\vec{C}$ an auxiliary field. When the magnetic field is redefined, the
+curl of a product must be carried out:
+```math
+\nabla\times \vec{H'} = \nabla\times (\vec{H} e^{-i(k_{x}x+k_{y}y)}) \tag{6}
+```
+so,
+```math
+\nabla\times \vec{H'} = e^{-i(k_{x}x+k_{y}y)} \nabla\times \vec{H} + \begin{pmatrix} -ik_{x} \\-ik_{y}\\0 \end{pmatrix} \ \times \vec{H'} \tag{7}
+```
+where the complex exponential in the second term has been absorbed by
+$\vec{H'}$. Substituting in equation (5) gives
+```math
+\nabla\times \vec{H'} = \vec{K'} = -i\omega (1+\frac{i\sigma_{D}}{\omega}) \vec{C'} + \begin{pmatrix} -ik_{x} \\-ik_{y}\\0 \end{pmatrix} \ \times \vec{H'}. \tag{8}
+```
+From here on in, the prime notation can be dropped since this applies to
+all fields. By introducing a new field $\vec{F}$, equation
+(8) can be written as
+```math
+\vec{K} = -i\omega(1+\frac{i\sigma_{D}}{\omega})\vec{C} - i\omega\vec{F}. \tag{9}
+```
+This new field satisfies the equation:
+```math
+\vec{F} = \vec{\bar{k}}\times\vec{H}, \tag{10}
+```
+where
+```math
+\vec{\bar{k}} =\frac{1}{\omega}\begin{pmatrix} k_{x} \\ k_{y}\\0 \end{pmatrix} \ =\begin{pmatrix} \sin{\theta}\cos{\phi} \\ \sin{\theta}\sin{\phi}\\0 \end{pmatrix} \ \tag{11}
```

Comment:
I think this equation is missing $n$ (the refractive index of the source medium) in the numerator.

Reply:
Thanks, I've added that now.